src/platform/web/ui/session/room/timeline/GapView.js

```diff
@@ -30,24 +30,18 @@ export class GapView extends TemplateView {
             isAtTop: vm => vm.isAtTop,
         };
         return t.li({ className }, [
-            t.map(vm => vm.isLoading,
-                (isLoading, t, vm) => {
-                    let elements;
-                    const error = vm.error;
-                    if (error) {
-                        elements = [t.strong(() => error.message)]; 
-                        if (error.showSpinner) {
-                            elements.unshift(spinner(t));
-                        }
-                    }
-                    else if (isLoading) {
-                        elements = [spinner(t), t.span(vm.i18n`Loading more messages …`)];
-                    }
-                    else {
-                        elements = t.span(vm.i18n`Not loading!`);
-                    }
-                    return t.div({ className: "GapView__container" }, elements);
-                })
+            t.if(vm => vm.showSpinner, (t) => spinner(t)),
+            t.span(vm => {
+                if (vm.error) {
```

Comment:
I'd usually put this kind of logic into a property on the vm rather than in the view.

Reply:
Resolved in https://github.com/vector-im/hydrogen-web/pull/834/commits/e6f43d6f4f32717153eda313a77a04ff970d2bbd